### PR TITLE
make batteries glow based on power contents (a redo that works)

### PR DESCRIPTION
**Pull request by Arkanic**
i went all the way as to add seperate icon sprites

as a fact, this is the fifth redesign to fix the anti aliasing

 now the holes in the battery are transparent, with the colour appearing behind them
the icon is an entirely seperate file of the old sprite

### CODE REVIEW

File: core/src/mindustry/world/blocks/power/Battery.java [hunk 1]
```diff
@@ -1,10 +1,41 @@
 package mindustry.world.blocks.power;
 
+import arc.*;
+import arc.graphics.Color;
+import arc.graphics.g2d.*;
+import mindustry.world.*;
+
 public class Battery extends PowerDistributor{
 
+    public TextureRegion baseRegion;
```
Comment (Anuken) [inline core/src/mindustry/world/blocks/power/Battery.java]: The base region is just a white square, so it can be drawn with `Fill.square()`.

**Comment by Anuken** [comment]
I'm going to merge this and implement some fixes on top myself.

**Comment by Arkanic** [comment]
nice
my first actual commit

File: core/src/mindustry/world/blocks/power/Battery.java (hunk 1 above)
Comment (Arkanic) [reply core/src/mindustry/world/blocks/power/Battery.java]: ok

**Pull request merged**